nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/service/canvas-utils.service.ts

```diff
@@ -956,7 +956,16 @@ export class CanvasUtils {
         return { x, y };
     }
 
+    public isClipboardAvailable(): boolean {
+        // system clipboard interaction requires the browser to be in a secured context.
+        return window.isSecureContext && Object.hasOwn(window, 'ClipboardItem');
```

Comment:
(outside of the scope for this PR): usually window is wrapped with an injection token for a few reasons:

testing
ssr support (which might be needed in the future)

Reply:
Good idea to inject window in the future, but there are many places where it is used directly now. We should address it across the board in its own commit. I filed this [1] JIRA, feel free to add commentary to it.

[1] [NIFI-14250](https://issues.apache.org/jira/browse/)

Reply:
Do we need similar logic in the `CopyDirective` so we don't show the copy button when copying it's supported?